### ISSUE TITLE
Issue: Escape commas in regex

### ISSUE THREAD

**Issue by aashayshah** [go-validator/validator#57]
There is a problem if a comma is a part of a the regex string. 
An example here:
```
type Company struct {
    Name string `validate:"regexp=^[a-zA-Z]{1,100}$"` 
}
```
This gives the error **"Name: unknown tag"** while doing `validator.Validate(C)`, C here is of type struct, of course.

The ([line](https://github.com/go-validator/validator/blob/v2/validator.go#L332)) in the file validator.go splits the tag. 

Escaping the commas and making modifications to the validator.go code would be the solution I think,

Any thoughts on how to proceed?

**Comment by dejjan** [comment]
Please fix, validation regex still many issue.

**Comment by jameshartig** [comment]
Fixed in #52. You'll need to escape the comma with 2 backslashes.

**Comment by kfirfer** [comment]
Hi @fastest963 

Tried with this:

```
type (
	Pagination struct {
		Page int    `query:"page" validate:"gt=0,lte=100000"`
		Size int    `query:"size" validate:"gt=0,lte=1000"`
		Sort string `query:"sort" validate:"max=30,^[a-zA-Z]{1\\,100}$"`
	}
)
```

And got this exception:

```
[PANIC RECOVER] Undefined validation function '^[a-zA-Z]{1\\' on field 'Sort' goroutine 39 [running]:\ngithub.com/labstack/echo/v4/middleware.RecoverWithConfig.func1.1.1(0x100f628, 0x400, 0x40000, 0x110b6c0, 0xc00070ae60)\n\t/home/vostro/go/pkg/mod/github.com/labstack/echo/v4@v4.1.17/middleware/recover.go:77 +0x16f\npanic(0xeae180, 0xc0003f5490)\n\t/home/vostro/sdk/go1.14.2/src/runtime/panic.go:975 +0x41b\ngithub.com/go-playground/validator/v10.(*Validate).parseFieldTagsRecursive(0xc00008a5a0, 0xc000703ac0, 0x19, 0xea2ce0, 0x4, 0xc000703ac7, 0xc, 0xeae100, 0xc00008b260, 0xc00008b2c0)\n\t/home/vostro/go/pkg/mod/github.com/go-playground/validator/v10@v10.4.0/cache.go:289 +0x1b79\ngithub.com/go-playground/validator/v10.(*Validate).extractStructCache(0xc00008a5a0, 0xf47ae0, 0xc000112ec0, 0x199, 0xe5b716, 0xa, 0x0)\n\t/home/vostro/go/pkg/mod/github.com/go-playground/validator/v10@v10.4.0/cache.go:150 +0x5fa\ngithub.com/go-playground/validator/v10.(*validate).validateStruct(0xc0000b7440, 0x10f9740, 0xc0000400e0, 0xe88980, 0xc000112ec0, 0x16, 0xf47ae0, 0xc000112ec0, 0x199, 0x1108f00,
```
